src/libbluechi/common/cfg.c

```diff
@@ -398,42 +398,48 @@ const char *cfg_dump(struct config *config) {
 }
 
 int cfg_agent_def_conf(struct config *config) {
-        int result = 0;
```

Comment:
Wouldn't this method be more readable if we use the opposite comparison?

```
        int result = cfg_set_default_section(config, CFG_SECT_AGENT);
        if (result == 0) {
            _cleanup_free_ char *default_node_name = get_hostname();
            result = cfg_set_value(config, CFG_NODE_NAME, default_node_name);
        }


        if (result == 0) {
            result = cfg_set_value(config, CFG_MANAGER_HOST, BC_DEFAULT_HOST);
        }

        if (result == 0) {
            result = cfg_set_value(config, CFG_HEARTBEAT_INTERVAL, AGENT_HEARTBEAT_INTERVAL_MSEC);
        }

        if (result == 0) {
            result = cfg_set_value(config, CFG_MANAGER_PORT, BC_DEFAULT_PORT);
        }

        if (result == 0) {
            const char *LOG_LEVEL_INFO = "INFO";
            result = cfg_set_value(config, CFG_LOG_LEVEL, LOG_LEVEL_INFO);
        }

        if (result == 0) {
            result = cfg_set_value(config, CFG_LOG_TARGET, BC_LOG_TARGET_STDERR);
        }

        if (result == 0) {
            result = cfg_set_value(config, CFG_LOG_IS_QUIET, NULL);
        }

        return result;
```

Reply:
I find it harder to read, but that might be because I am not used to this style. If you want, I can change it @mwperina

Reply:
Sure, up to you, it just seemed to me like too much unnecessary code  :-)

Reply:
Yes, this is unfortunately true for these kind of functions. I'd go with Ygals proposal, which seems like a nice compromise.

Reply:
Done :heavy_check_mark: